scripts/monthly_stats.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import boto3
+from app.config import Config
+from app.metrics.pennsieve import get_pennseive_download_metrics
+from scripts.monthly_downloads_html_template import create_html_template
+from scripts.email_sender import EmailSender
+import requests
+import datetime
+from dateutil.relativedelta import relativedelta
+
+class MonthlyStats(object):
+    def __init__(self, debug_mode=False, debug_email=''):
+        self.send_grid = EmailSender()
+        self.user_stats = {}
+        self.organization = Config.PENNSIEVE_ORGANIZATION
+        self._pennsieve_temp_api_key = ''
+        self.created_at = datetime.datetime.now()
+        self.run_day = 1  # This is the day of the month emails will be sent
+        self.debug_email = debug_email
+        self.debug_mode = debug_mode
+
+    # daily_run_check runs on a set day of the month. However, since we do nt want to send two emails to users if the
+    #       app restarts on the first day of the month, We assume an email has already been sent if the app has just
+    #       started and set a cooldown period of 24h. This ensures we pass the 1 day trigger for sending the emails
+    def daily_run_check(self):
+        now = datetime.datetime.now()
+        if now.day == self.run_day:  # Check if current day is run day
+            if (now - self.created_at) > datetime.timedelta(days=1):  # Do not run if app was started within 24h
+                self.run()
```

Comment:
I think there might be an issue if run_day is on the same day as created_at. I don't think this will ever get ran in that slim chance (or is that what the comment above is referring to about sending 2 emails if the app restarts? Bc in that case I do not see where the first one will get sent)

Reply:
> (or is that what the comment above is referring to about sending 2 emails if the app restarts? Bc in that case I do not see where the first one will get sent)
yeah you are right! 

`self.created_at` is set when sparc-api is run, which for production will be the last time the app is deployed on heroku.

Logic goes like this:
1. We want to send on one day every month
2. But if we restart the app on run day after the one day a month trigger is hit, we will hit the trigger and send again
3. To fix this we have a "cooldown" period of 24h after server start before sending any emails, ensuring if we redeploy on run day we wait until next month to send

Reply:
Ok that makes sense. Slack me the env vars you used so that I can verify sparc-api has the same ones. Once I verify they are set on sparc-api, I will merge this is in